src/services/lobbySocketService.ts

```
@@ -0,0 +1,134 @@
+import { io, Socket } from 'socket.io-client';
+import { GameUpdate } from '../types/GameTypes';
+
+export class LobbySocketService {
+  private socket: Socket | null = null;
+  private userId: string | null = null;
+  private gameId: string | null = null;
+  private callbacks: Map<string, Function[]> = new Map();
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)